scrunch/datasets.py

```diff
@@ -129,34 +113,23 @@ def get_dataset(dataset, connection=None, editor=False, project=None):
                 "Authenticate first with scrunch.connect() or by providing "
                 "config/environment variables")
     root = connection
-
-    if project:
-        if isinstance(project, six.string_types):
-            project_obj = get_project(project, connection)
-            ds = project_obj.get_dataset(dataset)
-        else:
-            ds = project.get_dataset(dataset)
-    else:
+    # search by dataset name
+    try:
+        shoji_ds = root.datasets.by('name')[dataset].entity
+    except KeyError:
+        # search by dataset id
         try:
-            shoji_ds = root.datasets.by('name')[dataset].entity
+            shoji_ds = root.datasets.by('id')[dataset].entity
         except KeyError:
+            # search by id on any project
             try:
-                shoji_ds = root.datasets.by('id')[dataset].entity
-            except KeyError:
-                try:
-                    dataset_url = urljoin(
-                        root.catalogs.datasets, '{}/'.format(dataset))
-                    shoji_ds = root.session.get(dataset_url).payload
-                except Exception:
-                    raise KeyError(
-                        "Dataset (name or id: %s) not found in context." % dataset)
-
-        ds = Dataset(shoji_ds)
-
-    if editor is True:
-        ds.change_editor(root.session.email)
-
-    return ds
+                dataset_url = urljoin(
+                    root.catalogs.datasets, '{}/'.format(dataset))
```

Comment:
It will, if the dataset ID passed lives on a project. This was requested, so no need to specify projects anymore.